crates/fuel-core/src/service/adapters/executor.rs

```diff
@@ -103,17 +103,21 @@ impl NewTxWaiterPort for NewTxWaiter {
 }
 
 impl PreconfirmationSenderPort for PreconfirmationSender {
-    async fn send(&self, preconfirmations: Vec<PreconfirmationStatus>) {
+    async fn send(&self, preconfirmations: Vec<Preconfirmation>) {
         // If the receiver is closed, it means no one is listening to the preconfirmations and so we can drop them.
         // We don't consider this an error.
-        let _ = self.sender.send(preconfirmations).await;
+        let _ = self.sender_signature_service.send(preconfirmations).await;
     }
 
-    fn try_send(
-        &self,
-        preconfirmations: Vec<PreconfirmationStatus>,
-    ) -> Vec<PreconfirmationStatus> {
-        match self.sender.try_send(preconfirmations) {
+    fn try_send(&self, preconfirmations: Vec<Preconfirmation>) -> Vec<Preconfirmation> {
+        for preconfirmation in &preconfirmations {
+            let tx_id = preconfirmation.tx_id;
+            let status = preconfirmation.status.clone();
+            self.tx_status_manager_adapter
+                .tx_status_manager_shared_data
+                .update_status(tx_id, status.into());
+        }
+        match self.sender_signature_service.try_send(preconfirmations) {
```

Comment:
If channel is full, you will call `update_status` for the same pre confirmation over and over

Reply:
No the try_send is outside of the loop.

Reply:
Fixed